src/helpers/ring.rs

```diff
@@ -0,0 +1,201 @@
+//!
+//! This module contains implementations and traits that enable MPC helpers to communicate with
+//! each other. In order for helpers to send messages, they need to know the destination. In some
+//! cases this might be the exact address of helper host/instance (for example IP address), but
+//! in many situations MPC helpers orchestrated into a "ring" - every helper instance has a peer
+//! on the right side and on the left side. They simply need to be able to send messages to the
+//! corresponding helper without needing to know the exact location - this is what this module
+//! enables MPC helper service to do.
+//!
+use crate::helpers::error::Error;
+use async_trait::async_trait;
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+use std::fmt::Debug;
+
+/// Trait for messages sent between helpers
+pub trait Message: Debug + Send + Serialize + DeserializeOwned + 'static {}
+
+impl<T> Message for T where T: Debug + Send + Serialize + DeserializeOwned + 'static {}
+
+/// Destination. Currently we only support Left and Right, but we could support the exact address
+/// too
+#[derive(Debug, PartialEq, Eq, Hash, Clone, Copy)]
+pub enum HelperAddr {
+    Left,
+    Right,
+}
+
+/// Trait for MPC helpers to communicate with each other. Helpers can send messages and
+/// receive messages from a specific helper.
+#[async_trait]
+pub trait Ring {
```

Comment:
I'll have to stop using the term "Ring" to refer to the concept in Mathematics: https://en.wikipedia.org/wiki/Ring_(mathematics)

That's fine - I think we don't need to draw any distinction with a field.